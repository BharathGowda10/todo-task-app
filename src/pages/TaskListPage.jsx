import React from "react";
import TaskList from "../components/TaskList";

const TaskListPage = () => {
  const tasks = [
    {
      id: 1,
      name: "Complete React project",
      date: "2023-10-01",
      time: "10:00 AM",
      type: "Work",
      priority: "High",
      completed: false,
    },
    {
      id: 2,
      name: "Grocery shopping",
      date: "2023-10-02",
      time: "5:00 PM",
      type: "Personal",
      priority: "Medium",
      completed: false,
    },
    {
      id: 3,
      name: "Gym workout",
      date: "2023-10-03",
      time: "7:00 AM",
      type: "Health",
      priority: "Low",
      completed: true,
    },
  ];
  return (
    <div>
      <TaskList tasks={tasks} />
    </div>
  );
};

export default TaskListPage;
