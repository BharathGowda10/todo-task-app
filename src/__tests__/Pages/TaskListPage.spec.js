import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import TaskListPage from '../../pages/TaskListPage';

const tasks = [
  {
    id: '1',
    name: 'Task One',
    dateTime: '2023-10-01T10:00',
    type: 'Work',
    priority: 'High',
  },
];

describe('TaskListPage', () => {
  it('matches snapshot', () => {
    const { asFragment } = render(
      <TaskListPage
        tasks={tasks}
        onDelete={jest.fn()}
        onEdit={jest.fn()}
        onCheckBoxClick={jest.fn()}
        fadingTasks={[]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
