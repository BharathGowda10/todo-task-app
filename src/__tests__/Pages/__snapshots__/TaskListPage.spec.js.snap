// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskListPage matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="row"
    >
      <div
        class="col-10 offset-1"
      >
        <ul
          class="list-group"
        >
          <li
            class="list-group-item border border-secondary rounded mb-2 "
          >
            <div
              class="row align-items-center"
            >
              <div
                class="col-1 d-flex justify-content-center"
              >
                <input
                  aria-label="Task completion checkbox"
                  class="form-check-input"
                  id="checkbox-1"
                  name="checkbox-1"
                  style="cursor: pointer; width: 1.5rem; height: 1.5rem;"
                  type="checkbox"
                />
              </div>
              <div
                class="col-5 d-flex flex-column"
              >
                <span
                  class="fw-bold fs-6 "
                >
                  Task One
                </span>
                <span
                  class="text-muted small"
                >
                  2023-10-01T10:00
                </span>
              </div>
              <div
                class="col-4 d-flex flex-column"
              >
                <span
                  class="badge fontsize bg-info text-dark mb-1"
                >
                  Work
                </span>
                <span
                  class="badge fontsize bg-danger"
                >
                  High
                </span>
              </div>
              <div
                class="col-2 d-flex justify-content-evenly align-items-center gap-1"
              >
                <svg
                  aria-label="edit-button"
                  data-testid="edit-button"
                  fill="currentColor"
                  height="30px"
                  role="button"
                  stroke="currentColor"
                  stroke-width="0"
                  style="cursor: pointer;"
                  viewBox="0 0 576 512"
                  width="30px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
                  />
                </svg>
                <svg
                  aria-label="delete-button"
                  data-testid="delete-button"
                  fill="currentColor"
                  height="25px"
                  role="button"
                  stroke="currentColor"
                  stroke-width="0"
                  style="cursor: pointer;"
                  viewBox="0 0 448 512"
                  width="25px"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
                  />
                </svg>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
